pkg/operator/ceph/file/mds/filesystem_test.go

```diff
@@ -0,0 +1,218 @@
+	var deploymentsUpdated *[]*apps.Deployment
+	mds.UpdateDeploymentAndWait, deploymentsUpdated = testopk8s.UpdateDeploymentAndWaitStub()
+
+	configDir, _ := ioutil.TempDir("", "")
+	// Output to check multiple filesystem creation
+	fses := `[{"name":"myfs","metadata_pool":"myfs-metadata","metadata_pool_id":1,"data_pool_ids":[2],"data_pools":["myfs-data0"]}]`
+
+	executor := &exectest.MockExecutor{
+		MockExecuteCommandWithOutputFile: func(debug bool, actionName string, command string, outFileArg string, args ...string) (string, error) {
+			return "{\"key\":\"mysecurekey\"}", nil
+		},
+		MockExecuteCommandWithOutput: func(debug bool, actionName string, command string, args ...string) (string, error) {
+			if strings.Contains(command, "ceph-authtool") {
+				cephtest.CreateConfigDir(path.Join(configDir, "ns"))
+			}
+
+			return "", nil
+		},
+	}
+	defer os.RemoveAll(configDir)
+	context := &clusterd.Context{
+		Executor:  executor,
+		ConfigDir: configDir,
+		Clientset: testop.New(3)}
+	fs := cephv1.CephFilesystem{
+		ObjectMeta: metav1.ObjectMeta{Name: "myfs", Namespace: "ns"},
+		Spec: cephv1.FilesystemSpec{
+			MetadataPool: cephv1.PoolSpec{Replicated: cephv1.ReplicatedSpec{Size: 1}},
+			DataPools:    []cephv1.PoolSpec{{Replicated: cephv1.ReplicatedSpec{Size: 1}}},
+			MetadataServer: cephv1.MetadataServerSpec{
+				ActiveCount: 1,
+				Resources: v1.ResourceRequirements{
+					Limits: v1.ResourceList{
+						v1.ResourceCPU: *resource.NewQuantity(100.0, resource.BinarySI),
+					},
+					Requests: v1.ResourceList{
+						v1.ResourceMemory: *resource.NewQuantity(1337.0, resource.BinarySI),
+					},
+				},
+			},
+		},
+	}
+	clusterInfo := &cephconfig.ClusterInfo{FSID: "myfsid"}
+
+	// start a basic cluster
+	err := createFilesystem(clusterInfo, context, fs, "v0.1", cephv1.CephVersionSpec{}, false, []metav1.OwnerReference{})
+	assert.Nil(t, err)
+	validateStart(t, context, fs)
+	assert.ElementsMatch(t, []string{}, testopk8s.DeploymentNamesUpdated(deploymentsUpdated))
+	testopk8s.ClearDeploymentsUpdated(deploymentsUpdated)
+
+	// starting again should be a no-op
+	err = createFilesystem(clusterInfo, context, fs, "v0.1", cephv1.CephVersionSpec{}, false, []metav1.OwnerReference{})
+	assert.Nil(t, err)
+	validateStart(t, context, fs)
+	assert.ElementsMatch(t, []string{"rook-ceph-mds-myfs-a", "rook-ceph-mds-myfs-b"}, testopk8s.DeploymentNamesUpdated(deploymentsUpdated))
+	testopk8s.ClearDeploymentsUpdated(deploymentsUpdated)
+
+	// Test multiple filesystem creation
+	executor = &exectest.MockExecutor{
+		MockExecuteCommandWithOutputFile: func(debug bool, actionName string, command string, outFileArg string, args ...string) (string, error) {
+			if contains(args, "ls") {
+				return fses, nil
+			}
+			return "{\"key\":\"mysecurekey\"}", errors.New("multiple fs")
+		},
+	}
+	context = &clusterd.Context{
+		Executor:  executor,
+		ConfigDir: configDir,
+		Clientset: testop.New(3)}
+
+	//Create another filesystem which should fail
```

Comment:
Please have a space at the beginning of the comment as the other comments have (also lowercase begin, though I would vote for uppercase first char).